### PR TITLE
Make displays respect PauliString coefficient

### PR DESCRIPTION
**Pull request by mpharrigan**
This one was a real head scratcher!

### CODE REVIEW

**Comment by mpharrigan** [comment]
bump! I consider this a bug

File: cirq/ops/display_test.py [hunk 1]
```diff
@@ -199,6 +235,21 @@ def test_approx_pauli_string_expectation_value(measurements, value):
...
+@pytest.mark.parametrize('measurements, value, coefficient', [
+    (np.array([[0, 0, 0], [0, 0, 0]]), 1, 0.123),
+    (np.array([[0, 0, 0], [0, 0, 1]]), 0, 999),
+    (np.array([[0, 1, 0], [1, 0, 0]]), -1, -1),
+    (np.array([[0, 1, 0], [1, 1, 1]]), -1, 1),
+])
+def test_approx_pauli_string_expectation_value_with_coef(
+        measurements, value, coefficient):
+    display = cirq.pauli_string_expectation(cirq.PauliString(
+        {}, coefficient=coefficient),
```
Comment (dabacon) [inline cirq/ops/display_test.py]: wonky formatting

Comment (mpharrigan) [reply cirq/ops/display_test.py]:
```
$ check/format-incremental 
Comparing against revision 'origin/master'.

Changes in cirq/ops/display_test.py require formatting:
--- cirq/ops/display_test.py    (original)
+++ cirq/ops/display_test.py    (reformatted)
@@ -243,8 +243,9 @@
 ])
 def test_approx_pauli_string_expectation_value_with_coef(
         measurements, value, coefficient):
-    display = cirq.pauli_string_expectation(
-        cirq.PauliString({}, coefficient=coefficient), num_samples=1)
+    display = cirq.pauli_string_expectation(cirq.PauliString(
+        {}, coefficient=coefficient),
+                                            num_samples=1)
     assert display.value_derived_from_samples(measurements) == \
            value * coefficient

```

Comment (Strilanc) [reply cirq/ops/display_test.py]: The formatter always knows best, even when it doesn't.

**Comment by mpharrigan** [comment]
Even if we delete the displays as such, we'll always need a function for computing an expectation value of a `PauliString` from a wavefunction or density matrix

**Comment by Strilanc** [comment]
That sounds like an excellent method to have on PauliString itself

**Pull request merged**